plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -300,12 +301,15 @@ void deleteBlob(String blobName) throws IOException {
      *
      * @param pathStr Name of path to delete
      */
-    void deleteDirectory(String pathStr) throws IOException {
+    void deleteDirectory(String pathStr, LongConsumer resultConsumer) throws IOException {
         SocketAccess.doPrivilegedVoidIOException(() -> {
             Page<Blob> page = client().get(bucketName).list(BlobListOption.prefix(pathStr));
             do {
                 final Collection<String> blobsToDelete = new ArrayList<>();
-                page.getValues().forEach(b -> blobsToDelete.add(b.getName()));
+                page.getValues().forEach(b -> {
+                    blobsToDelete.add(b.getName());
+                    resultConsumer.accept(b.getSize());
```

Comment:
I think now hat we return the delete stats instead of updating some mutable object this is fine. I can see an argument on GCS, that you technically could separate the case of deleting and already missing blob out (not on S3 though) to get more exact counts but I figured that might not be worth the complexity (we could do that in a follow up though, there's some simplifications API wise here that I had planned that would make it easier to do that, but here it seemed like needless added noise)?

Reply:
Good for me, thanks